5-AccessControl/1-call-api-roles/SPA/src/components/RouteGuard.jsx

```diff
@@ -1,40 +1,36 @@
-import React, { useState, useEffect } from "react";
-import { Route } from "react-router-dom";
-import { useMsal } from "@azure/msal-react";
-
-export const RouteGuard = ({ Component, ...props }) => {
+import React, { useState, useEffect } from 'react';
+import { useMsal } from '@azure/msal-react';
 
+export const RouteGuard = ({ ...props }) => {
     const { instance } = useMsal();
     const [isAuthorized, setIsAuthorized] = useState(false);
+    const currentAccount = instance.getActiveAccount();
 
     const onLoad = async () => {
-        const currentAccount = instance.getActiveAccount();
 
         if (currentAccount && currentAccount.idTokenClaims['roles']) {
-            let intersection = props.roles
-                .filter(role => currentAccount.idTokenClaims['roles'].includes(role));
-
+            let intersection = props.roles.filter((role) => currentAccount.idTokenClaims['roles'].includes(role));
             if (intersection.length > 0) {
                 setIsAuthorized(true);
+            }else{
+                setIsAuthorized(false);
             }
         }
-    }
+    };
 
     useEffect(() => {
         onLoad();
-    }, [instance]);
+    }, [instance, currentAccount]);
 
     return (
         <>
-            {
-                isAuthorized
-                    ?
-                    <Route {...props} render={routeProps => <Component {...routeProps} />} />
-                    :
-                    <div className="data-area-div">
-                        <h3>You are unauthorized to view this content.</h3>
-                    </div>
-            }
+            {isAuthorized ? (
+                <div>{props.children}</div>
+            ) : (
+                <div className="data-area-div">
+                    <h3>You are unauthorized to view this content.</h3>
```

Comment:
>You are unauthorized to view this content [](http://example.com/codeflow?start=24&length=41)

You are unauthorized to view this content. Please assign yourself to the role "TaskAdmin" to view data in this screen